MobileWallet/Common/Views/Secure View/SecureWrapperView.swift

```diff
@@ -0,0 +1,130 @@
+//  SecureWrapperView.swift
+
+/*
+	Package MobileWallet
+	Created by Adrian Truszczyński on 13/02/2024
+	Using Swift 5.0
+	Running on macOS 14.2
+
+	Copyright 2019 The Tari Project
+
+	Redistribution and use in source and binary forms, with or
+	without modification, are permitted provided that the
+	following conditions are met:
+
+	1. Redistributions of source code must retain the above copyright notice,
+	this list of conditions and the following disclaimer.
+
+	2. Redistributions in binary form must reproduce the above
+	copyright notice, this list of conditions and the following disclaimer in the
+	documentation and/or other materials provided with the distribution.
+
+	3. Neither the name of the copyright holder nor the names of
+	its contributors may be used to endorse or promote products
+	derived from this software without specific prior written permission.
+
+	THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+	CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+	INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+	OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+	DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+	CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+	SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+	NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+	LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+	HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+	CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+	OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+	SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import TariCommon
+import Combine
+
+enum ScreenshotPreventionStatus {
+    case enforced
+    case turnedOff
+    case automatic
+}
+
+final class SecureWrapperView<MainView: UIView>: UIView {
+
+    // MARK: - Subviews
+
+    @View private(set) var view: MainView
+
+    @View private var textField: UITextField = {
+        let view = UITextField()
+        view.isUserInteractionEnabled = false
+        view.backgroundColor = .clear
+        return view
+    }()
+
+    private var contentView: UIView? { textField.subviews.first { type(of: $0).description().contains("CanvasView") }}
+
+    // MARK: - Properties
+
+    @Published var screenshotPreventionStatus: ScreenshotPreventionStatus = .automatic
+    private var cancellables = Set<AnyCancellable>()
+
+    // MARK: - Initialisers
+
+    convenience init() {
+        self.init(mainView: MainView())
+    }
+
+    init(mainView: MainView) {
+        self.view = mainView
+        super.init(frame: .zero)
+        setupViews()
+        setupCallbacks()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
```

Comment:
Isn't it a placeholder for the function? Is it implemented?

Reply:
This method is required when you inherit from UIVIew but it is only called when the view is decoded from the nib/xib or storyboards. We don't use them in our project so that method can stay unimplemented.